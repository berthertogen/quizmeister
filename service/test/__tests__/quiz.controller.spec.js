let roundIds;
let shortId;

test("should create quiz on Post", async () => {
  const responseRound = await buildPostRequest("round", "", roundFactory.create([]));
  roundIds = [responseRound.body.roundId];
  const response = await buildPostRequest("quiz", "", quizFactory.create(roundIds));
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(true));
});

test("should get quiz on Get", async () => {
  const response = await buildGetRequest('quiz', '1');
  shortId = response.body.shortId;
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(true));
});

test("should get quiz on Get (shortId)", async () => {
  const response = await buildGetRequest('quiz', `short/${shortId}`);
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(true));
});

test("should get quiz on Get (search)", async () => {
  const response = await buildGetRequest('quiz', 'search/0/20');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot([quizFactory.ignoreProperties(true)]);
});

test("should update quiz on Put", async () => {
  const response = await buildPutRequest("quiz", "1", quizFactory.update(roundIds));
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(true));
});

test("should update quiz on Put (status)", async () => {
  const response = await buildPutRequest("quiz", "1/status/2", {});
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(true));
});

test("should get quiz on Get (open)", async () => {
  const response = await buildGetRequest('quiz', 'open/0/20');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot([quizFactory.ignoreProperties(true)]);
});

test("should copy quiz on Put (copy)", async () => {
  const response = await buildPutRequest("quiz", "1/copy", []);
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(quizFactory.ignoreProperties(false));
});

test("should delete quiz on Delete", async () => {
  const responseDelete1 = await buildDeleteRequest("quiz", "1/true");
  assertContentTypeJsonAndStatus200(responseDelete1);
  const responseDelete2 = await buildDeleteRequest("quiz", "2/true");
  assertContentTypeJsonAndStatus200(responseDelete2);
  const responseGetQuizzes = await buildGetRequest('quiz', 'search/0/20');
  expect(responseGetQuizzes.status).toBe(200);
  expect(responseGetQuizzes.body).toMatchSnapshot();
  const responseGetRounds = await buildGetRequest('round', 'search/0/20');
  expect(responseGetRounds.status).toBe(200);
  expect(responseGetRounds.body).toMatchSnapshot();
});