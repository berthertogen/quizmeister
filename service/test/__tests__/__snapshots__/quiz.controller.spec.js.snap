// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should copy quiz on Put (copy) 1`] = `
Object {
  "date": "2100-01-31T23:00:00",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 2,
  "remark": "remark",
  "rounds": Array [],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 1,
  "subscriptions": null,
  "title": "updated title - copy",
}
`;

exports[`should create quiz on Post 1`] = `
Object {
  "date": "2100-01-31T23:00:00Z",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 1,
  "remark": "remark",
  "rounds": Array [
    Object {
      "modifiedOn": Any<String>,
      "questions": null,
      "remark": "remark",
      "roundId": 1,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "theme": "theme",
      "title": "title",
    },
  ],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 1,
  "subscriptions": null,
  "title": "title",
}
`;

exports[`should delete quiz on Delete 1`] = `Array []`;

exports[`should delete quiz on Delete 2`] = `Array []`;

exports[`should get quiz on Get (open) 1`] = `
Array [
  Object {
    "date": "2100-01-31T23:00:00",
    "location": "location",
    "maxSubscriptions": 20,
    "modifiedOn": Any<String>,
    "quizId": 1,
    "remark": "remark",
    "rounds": Array [
      Object {
        "modifiedOn": Any<String>,
        "questions": null,
        "remark": "remark",
        "roundId": 1,
        "searchField": Any<String>,
        "shortId": Any<String>,
        "theme": "theme",
        "title": "title",
      },
    ],
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 2,
    "subscriptions": Array [],
    "title": "updated title",
  },
]
`;

exports[`should get quiz on Get (search) 1`] = `
Array [
  Object {
    "date": "2100-01-31T23:00:00",
    "location": "location",
    "maxSubscriptions": 20,
    "modifiedOn": Any<String>,
    "quizId": 1,
    "remark": "remark",
    "rounds": Array [
      Object {
        "modifiedOn": Any<String>,
        "questions": null,
        "remark": "remark",
        "roundId": 1,
        "searchField": Any<String>,
        "shortId": Any<String>,
        "theme": "theme",
        "title": "title",
      },
    ],
    "searchField": Any<String>,
    "shortId": Any<String>,
    "status": 1,
    "subscriptions": Array [],
    "title": "title",
  },
]
`;

exports[`should get quiz on Get (shortId) 1`] = `
Object {
  "date": "2100-01-31T23:00:00",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 1,
  "remark": "remark",
  "rounds": Array [
    Object {
      "modifiedOn": Any<String>,
      "questions": null,
      "remark": "remark",
      "roundId": 1,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "theme": "theme",
      "title": "title",
    },
  ],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 1,
  "subscriptions": Array [],
  "title": "title",
}
`;

exports[`should get quiz on Get 1`] = `
Object {
  "date": "2100-01-31T23:00:00",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 1,
  "remark": "remark",
  "rounds": Array [
    Object {
      "modifiedOn": Any<String>,
      "questions": null,
      "remark": "remark",
      "roundId": 1,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "theme": "theme",
      "title": "title",
    },
  ],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 1,
  "subscriptions": Array [],
  "title": "title",
}
`;

exports[`should update quiz on Put (status) 1`] = `
Object {
  "date": "2100-01-31T23:00:00",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 1,
  "remark": "remark",
  "rounds": Array [
    Object {
      "modifiedOn": Any<String>,
      "questions": null,
      "remark": "remark",
      "roundId": 1,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "theme": "theme",
      "title": "title",
    },
  ],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 2,
  "subscriptions": Array [],
  "title": "updated title",
}
`;

exports[`should update quiz on Put 1`] = `
Object {
  "date": "2100-01-31T23:00:00Z",
  "location": "location",
  "maxSubscriptions": 20,
  "modifiedOn": Any<String>,
  "quizId": 1,
  "remark": "remark",
  "rounds": Array [
    Object {
      "modifiedOn": Any<String>,
      "questions": null,
      "remark": "remark",
      "roundId": 1,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "theme": "theme",
      "title": "title",
    },
  ],
  "searchField": Any<String>,
  "shortId": Any<String>,
  "status": 1,
  "subscriptions": null,
  "title": "updated title",
}
`;
